src/libraries/System.Buffers/tests/ArrayPool/CollectionTests.cs

```diff
@@ -140,7 +142,9 @@ private static bool ValidateTrimState(object pool, string trimString)
             return parsedTrim;
         }
 
-        [ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsPreciseGcSupported))]
+        private static bool IsPreciseGcSupportedAndRemoteExecutorSupported => PlatformDetection.IsPreciseGcSupported && RemoteExecutor.IsSupported;
```

Comment:
I wonder if we should expose a property in `PlatformDetection` maybe using Reflection to simplify this cases. 

We could also define a compile constant if `IncludeRemoteExecutor` is set to true, and just expose that property when the constant is defined. 

https://github.com/dotnet/runtime/blob/617692461d55b7ae8162138e572f280e4c1df5f5/eng/testing/xunit/xunit.props#L9

Reply:
What are you trying to simplify? To me this already looks simple enough.

Reply:
I think what @safern meant was that if we had a property on `PlatformDetection` you could simply write the `ConditionalTheory` like this since it accepts multiple parameters:

```
[ConditionalTheory(typeof(PlatformDetection), nameof(PlatformDetection.IsPreciseGcSupported), nameof(PlatformDetection.IsRemoteExecutorSupported))]
```

Reply:
I was trying to simplify the case were we have multiple conditionals coming from a different type, so that you don't have to introduce a property to do the evaluation and then reference that property, but I think it is not a big deal. Just laying out the options, anyway it looks good to me as is.